### ISSUE TITLE
Nonstandard keyboard function in settings menu

### ISSUE THREAD

**Issue by Woebin** [Chocobozzz/PeerTube#7135]
### Describe the current behavior

These are not formally accessibility issues, but rather opportunities for improved keyboard interaction found during an accessibility review.

The video player's settings menu has several oddities when it comes to keyboard interaction:
![The opened settings menu and video quality "main" menu item.](https://github.com/user-attachments/assets/ec94aca2-be2b-476a-ab37-24f2b3d25271)

- It's not possible to close the settings menu inside the player with `Esc`.
- The video quality submenu doesn't open when the current setting is in focus and the right arrow key is pressed.
- If the settings menu is opened by mouse click, it doesn't receive keyboard focus as expected.

### Describe the expected behavior

Menus are expected to be closeable using the `Esc` key, submenus with an associated arrow icon pointing to the right should open when the right arrow key is pressed while the parent has focus, and if the mouse is used to open the menu keyboard focus is expected to follow.

Using standard elements with appropriate `role` and `aria` attributes and ensuring that keyboard operation works as expected would resolve these issues.
- For the settings menu, add an `id` value. Add `aria-controls="[settings-menu-id]"` to the settings button (the cog icon).
- Ensure that the quality options list is a proper submenu of the quality options in the settings menu.

### Additional information

* PeerTube instance:
  * URL: https://peertube-eu.howlround.com/
  * Version: 7.2.1
  * NodeJS version: Unknown
  * Ffmpeg version: Unknown

* Browser name, version and platforms on which you could reproduce the bug: MacOS 15.5, Google Chrome Version 137.0.7151.120 (Official Build) (arm64)

**Additional context**
This issue was recorded as part of an accessibility audit carried out by [Axess Lab](https://axesslab.com/), contracted by [HowlRound Theatre Commons](https://howlround.com/).

**Comment by Chocobozzz** [comment]
Thanks! Should be fixed by https://github.com/Chocobozzz/PeerTube/commit/27f197692954b5763404134e44b9636dc8c3a36f https://github.com/Chocobozzz/PeerTube/commit/2011ea37e620f63da395de5089998fd767452c7d  https://github.com/Chocobozzz/PeerTube/commit/91002ac0422a40ceb724d04bae7d338989687cd9 and https://github.com/Chocobozzz/PeerTube/commit/faeacaec6831802a8599f0f044eb82240e9cfc82
